text/0000-extended-standard-library.md

```diff
@@ -0,0 +1,370 @@
+
+It should be OK to remove previously included functionality if doing so makes sense. This is part of evolving APIs, and should be regarded as potentially healthy as opposed to necessarily reflective of a failure.
+
+## Adding Functionality
+
+Once a decision is made to try to include functionality in the ESL, the ESL community should try to find a suitable existing crate with maintainers who are willing to join the ESL.
+
+If a maintainer chooses to move their crate into the ESL, doing so does not require them to relinquish their role as a maintainer. They do need to commit to upholding the policies that the ESL community has decided on. In order to help with that and other maintenance tasks, maintainers will receive various forms of support as part of ESL membership (see below).
+
+If the ESL community would like to add functionality but is unable to find a suitable crate with maintainers willing to join, the community will have to consider their options, which include creating a new crate or not adding the functionality. If the functionality is required in order to enable the addition of higher-order functionality, an exception could be made to allow for a dependency outside of the ESL.
+
+## Maintainer Support
+
+The ESL should endeavor to provide the following forms of support to maintainers:
+
+* **Infrastructure Assistance**. Testing, security, release, and benchmarking infrastructure managed by an ESL infrastructure partner.
+* **Funding**. The opportunity to receive grants to help cover the cost of their contributions.
+* **Security Release Coverage**. If maintainers are unavailable to respond to a security issue, there is a team that can back them up.
+* **Legal Assistance**. Legal assistance as necessary from an ESL legal partner.
+
+## Crate Names
+
+All crates in the ESL would ideally have names that:
+
+1) Clearly describe the functionality they offer
+2) Contain a reliable indicator of ESL membership
+
+This would help Rust developers to identify functionality, understand where their dependencies are coming from, and make their Rust code easier to read.
+
+However, the amount of effort involved in renaming existing popular crates and a lack of namespacing ([Rust RFC 3243](https://rust-lang.github.io/rfcs/3243-packages-as-optional-namespaces.html) remains unimplemented) makes this ideal difficult to achieve.
+
+As such, pre-existing crates that join the ESL will keep their names unless there is a compelling reason to make a change.
+
+If a crate is created or renamed for the ESL, a name clearly describing its functionality will be chosen.
+
+Crate users will have to confirm ESL membership by checking the GitHub organization.
+
+## Process for Creating the ESL
+
+1. Set Up ESL Governance With Initial Crate Maintainers
+
+A successful ESL depends on maintainers of existing crates wanting to move their crates into the ESL. This may not be the case for every piece of functionality, but it will need to be the case for much of it.
```

Comment:
This is a real tension where I have to decide between
- Being blessed or fading into obscurity
- The policies if they are too heavy